v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/dialectadapter/mysql/MysqlDialectAdapter.java

```diff
@@ -374,6 +374,28 @@ private ImmutableMap<String, SourceColumnType> getTableCols(
 
   private ImmutableSet<String> binaryColumnTypes = ImmutableSet.of("BINARY", "VARBINARY", "BLOB");
 
+  /**
+   * Get the PadSpace attribute from {@link ResultSet} for index discovery query {@link
+   * #getIndexDiscoveryQuery(SourceSchemaReference)}. This method takes care of the fact that older
+   * versions of MySQL notably Mysql5.7 don't have a {@link
+   * InformationSchemaStatsCols#PAD_SPACE_COL} column and default to PAD SPACE comparisons.
+   */
+  @VisibleForTesting
+  @Nullable
+  protected String getPadSpaceString(ResultSet resultSet) throws SQLException {
+    ResultSetMetaData metaData = resultSet.getMetaData();
+    for (int i = 0; i < metaData.getColumnCount(); i++) {
+      if (metaData.getColumnName(i + 1).equals(InformationSchemaStatsCols.PAD_SPACE_COL)) {
```

Comment:
a cleaner way would be to fetch by column name if it exists. But not a big deal either way.